File: number_of_zero-filled_subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        res = 0
        count = 0

        for i in range(len(nums)):
            if nums[i] == 0:
                count += 1
            else:
                count = 0
            
            res += count
        
        return res

File: number_of_zero-filled_subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long res = 0;
        int count = 0;

        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 0) {
                count += 1;
            } else {
                count = 0;
            }
            res += count;
        }

        return res;        
    }
};
